docs/tutorials/batch_showcase.ipynb

```diff
@@ -0,0 +1,578 @@
+    "my_bounds([1,2,3,4,5])"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As in case 1, we're able to pass this function and the \"flux\" column along to run the function. However, this time we need the `meta` to be set. The `meta` is a needed component of `Dask's` lazy evaluation. As `Dask` does not actually compute results until requested to, `meta` serves as the expected form of the output. In this case, we just need to let `Dask` know that a min and max column will be present in a dataframe (TAPE will always return a dataframe)  and that both will be float values.\n",
+    "\n",
+    "For more information on the `Dask` meta argument, read their [documentation](https://blog.dask.org/2022/08/09/understanding-meta-keyword-argument)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Default Batch\n",
+    "\n",
+    "res1 = ens.batch(my_bounds, \"flux\", meta={\"min\":float, \"max\":float}) # Requires meta to be set\n",
+    "res1.compute()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The same flexibility with grouping extends to case 2, with again needing to specify the `meta`. Note that the meta given to `Ensemble.batch` remains the same, only depending on the function output, it handles the meta for any columns generated by the grouping on it's own."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Multi-level groupbys, note that meta does not need to change\n",
+    "res2 = ens.batch(my_bounds, \"flux\", on=[\"id\", \"band\"], meta={\"min\":float, \"max\":float}) # Requires meta to be set\n",
+    "res2.compute()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the `by_band` kwarg extends the output columns to be per-band."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Using by_band\n",
+    "\n",
+    "res3 = ens.batch(my_bounds, \"flux\", by_band=True, meta={\"min\":float, \"max\":float}) # Requires meta to be set\n",
+    "res3.compute()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Case 3: Functions That Return a DataFrame"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we define a function, `my_bounds_df` that computes the same quantities as `my_bounds` above, but in this case we return a dataframe of the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def my_bounds_df(flux):\n",
+    "    return pd.DataFrame({'min':[np.min(flux)], 'max':[np.max(flux)]})\n",
```

Comment:
I might be confused here but am wondering if returning 

```
    return pd.Series([np.min(flux), np.max(flux)], index=['min', 'max'])
```

resolves some of the weird re-indexing issue. It's a bit counter-intuitive, but this is what's recommended in the [`pd.DataFrame.apply` docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.apply.html)

Reply:
I'm assuming you mean to return a dataframe here? Returning a series with the index set does resolve this, but that's basically what case 2 shows. For users who have some pre-established functionality that returns a dataframe, the issue seems to persist even if you try to set the index as in that example (see attached). At least, that's what I'm trying to show with case 3

![Screen Shot 2023-12-19 at 2 50 55 PM](https://github.com/lincc-frameworks/tape/assets/8853505/59fa1e71-cd04-44af-91e9-fa9601b792b1)

Reply:
Ah that makes sense to me!